stl/inc/chrono

```diff
@@ -619,6 +671,1240 @@ namespace chrono {
+        return year{_Year} / _Mwdl;
+    }
+    _NODISCARD constexpr year_month_weekday_last operator/(
+        const month_weekday_last& _Mwdl, const year& _Year) noexcept {
+        return _Year / _Mwdl;
+    }
+    _NODISCARD constexpr year_month_weekday_last operator/(const month_weekday_last& _Mwdl, int _Year) noexcept {
+        return year{_Year} / _Mwdl;
+    }
+
+    // Calendrical constants
+    inline constexpr weekday Sunday{0};
+    inline constexpr weekday Monday{1};
+    inline constexpr weekday Tuesday{2};
+    inline constexpr weekday Wednesday{3};
+    inline constexpr weekday Thursday{4};
+    inline constexpr weekday Friday{5};
+    inline constexpr weekday Saturday{6};
+
+    inline constexpr month January{1};
+    inline constexpr month February{2};
+    inline constexpr month March{3};
+    inline constexpr month April{4};
+    inline constexpr month May{5};
+    inline constexpr month June{6};
+    inline constexpr month July{7};
+    inline constexpr month August{8};
+    inline constexpr month September{9};
+    inline constexpr month October{10};
+    inline constexpr month November{11};
+    inline constexpr month December{12};
+
+    constexpr unsigned _Width(intmax_t _Num, const intmax_t _Den) {
+        _STL_ASSERT(_Den > 0, "Denominator can't be less than 1.");
+        unsigned _Result = 0;
+        // doing everything in for > using a while loop
+        for (; _Num % _Den != 0 && _Result < 19; _Num = _Num % _Den * 10, ++_Result) {
+        }
+        return _Result == 19 ? 6 : _Result;
+    }
+
+    constexpr intmax_t _Pow10(const int _Exp) {
+        if (_Exp == 0) {
+            return 1;
+        }
+        return 10 * _Pow10(_Exp - 1);
+    }
+
+    template <class _Duration>
+    requires _Is_duration_v<_Duration> class hh_mm_ss {
+    public:
+        static constexpr unsigned fractional_width = _Width(_Duration::period::num, _Duration::period::den);
+        using precision =
+            duration<common_type_t<typename _Duration::rep, seconds::rep>, ratio<1, _Pow10(fractional_width)>>;
+
+        constexpr hh_mm_ss() noexcept : hh_mm_ss{_Duration::zero()} {}
+        // clang-format off
+        constexpr explicit hh_mm_ss(_Duration _Dur)
+            : _Is_neg{_Dur < _Duration::zero()},
+              _Hours{duration_cast<chrono::hours>(_STD chrono::abs(_Dur))},
+              _Mins{duration_cast<chrono::minutes>(_STD chrono::abs(_Dur) - hours())},
+              _Secs{duration_cast<chrono::seconds>(_STD chrono::abs(_Dur) - hours() - minutes())} {
+            // clang-format on
+            if constexpr (treat_as_floating_point_v<typename precision::rep>) {
+                _Sub_secs = _STD chrono::abs(_Dur) - hours() - minutes() - seconds();
+            } else {
+                _Sub_secs = duration_cast<precision>(_STD chrono::abs(_Dur) - hours() - minutes() - seconds());
+            }

```

Comment:
Why can't it just be
```suggestion
            _Sub_secs = duration_cast<precision>(_STD chrono::abs(_Dur) - hours() - minutes() - seconds());
```

Reply:
🤷‍♂️ was just following what the standard says https://eel.is/c++draft/time.hms#members-3.5